atlp-rwanda/winners-c8-bn-be
# #[N/A] Documents Signup, Signin and email verification endpoints in Swagger

#### What does this PR do?
This PR sets up swagger documentation for signup, signin and  email verification endpoints. It also has a commit that changes one of the email verification test to pass when an email is sent using a fake sendgrid API (for the sake of avoiding charges when tests are run).
#### Description of Task to be completed?
Have the swagger documentation for these endpoints:
`POST /api/auth/register/`
`POST /api/auth/signin`
`GET http://localhost:5000/api/auth/register/verifyuser/:token`
#### How should this be manually tested?
Step 1. clone this repository
Step 2. go to this branch by running `git checkout ch-swagger-signup-emailverification`
Step 3. run `npm install`
Step 4. make sure all the required `.env.example` variables are set, and run `npm run dev`
Step 5. Access the swagger docs on http://localhost:5000/api-docs/
#### Any background context you want to provide?
As mentioned above, this PR also has a commit that changes one of the email verification test to pass when an email is sent using a fake sendgrid API (for the sake of avoiding charges when tests are run). Therefore, when it's merged with develop, sending real emails will require a valid sendgrid api key will shall be charged.
#### What are the relevant pivotal tracker stories?
#182210043 , #182210046 , #182210044
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A
